tests/worker/dependency_manager_test.py

```diff
@@ -0,0 +1,116 @@
+import json
+import os
+import tempfile
+import threading
+import time
+import unittest
+
+from worker.dependency_manager import DependencyManager
+from worker.file_util import remove_path
+
+class DependencyManagerTest(unittest.TestCase):
+    def setUp(self):
+        self.work_dir = tempfile.mkdtemp()
+        self.manager = DependencyManager(self.work_dir, None)
+
+    def tearDown(self):
+        remove_path(self.work_dir)
+
+    def test_load_state(self):
+        self.manager.finish_run('uuid1')
+        self.manager.finish_run('uuid2')
+        with open(os.path.join(self.work_dir, 'random_file'), 'w'):
+            pass
+        self.assertIn('random_file', os.listdir(self.work_dir))
+        new_manager = DependencyManager(self.work_dir, 1)
+        self.check_state([('uuid1', ''), ('uuid2', '')], new_manager)
+        self.assertIn('state.json', os.listdir(self.work_dir))
+        self.assertNotIn('random_file', os.listdir(self.work_dir))
+
+    def test_downloading(self):
+        self.manager.add_dependee('uuid1', '', 'uuid2')
```

Comment:
Same comment as I had for your other PR: dependee is not a real word and I find it confusing.